srtcore/sync.h

```diff
@@ -938,6 +939,7 @@ class CGlobEvent
 // CThread class
 //
 ////////////////////////////////////////////////////////////////////////////////
+#define resetThread(th) do { new (&th) CThread; } while(0)
```

Comment:
You can do it as well without PP.
(This `(void)` is only to a-void compiler warnings)

```suggestion
inline void resetThread(CThread& th) { (void)new (&th) CThread; }
```